### PR TITLE
Connecting Game State to UI

### PR DESCRIPTION
**Pull request by nmoadev**
IGNORE: nmoadev.full-fix. This PR covers that branch, but resolves conflicts

* Load models from glb files
* Click on mesh to open "project selection" window
* Fancy new next turn button
* Lots of state plumbing through the app

PENDING ISSUES
* only the last position for a building seems to be respected
* Camera resets on state change

### CODE REVIEW

File: src/components/view/MapLocation.tsx [hunk 1]
```diff
@@ -1,20 +1,25 @@
-import { useRef, useState } from 'react'
+
+import { useLoader } from '@react-three/fiber'
+import { useState } from 'react'
+import { GLTFLoader } from 'three/examples/jsm/loaders/GLTFLoader'
+import { Tile } from '../game_logic'
```
Comment (hypevhs) [inline src/components/view/MapLocation.tsx]: You have several invalid paths (check the linter)

File: src/do-three-js.ts [hunk 2]
```diff
@@ -0,0 +1,93 @@
+import * as THREE from 'three'
+import { GLTFLoader } from 'three/examples/jsm/loaders/GLTFLoader'
+import { MapControls } from 'three/examples/jsm/controls/OrbitControls'
+
+export function doThreeJs () {
+  // Select the canvas from the document
+  const canvReference = document.getElementById('threejscanvas')
```
Comment (hypevhs) [inline src/do-three-js.ts]: What is this file for?

**Review by hypevhs — CHANGES_REQUESTED** [review]
Re: camera reset, I opened #15

**Review by scottysseus — APPROVED** [review]
We can keep iterating on the reducers. I can see those growing large enough to warrant their own file(s), for example. Looks awesome so far!

File: src/do-three-js.ts (hunk 2 above)
Comment (scottysseus) [reply src/do-three-js.ts]: We can probably get rid of it later; I'll keep it around for now

File: src/components/GameDisplay.tsx [hunk 3]
```diff
@@ -1,11 +1,57 @@
...
+
+  const [gameState, dispatch] = useReducer(reducer, startingState)
+  const [uiState, dispatchUI] = useReducer(uiReducer, startingUiState)
+
   return (
     <div className='container'>
-      <GameViewPort />
-      <Hud />
+      <GameViewPort gameState={gameState} uiState={uiState} dispatchUi={dispatchUI} />
+      {/* eslint-disable-next-line react/jsx-pascal-case */}
```
Comment (hypevhs) [inline src/components/GameDisplay.tsx]: This can be removed

File: src/components/view/Map.tsx [hunk 4]
```diff
@@ -1,16 +1,42 @@
+import { Suspense } from 'react'
+import { MapState } from '../game_logic'
+import { uiDispatcher, UiState } from './GameDisplay'
```
Comment (hypevhs) [inline src/components/view/Map.tsx]: These are the other two invalid paths I was mentioning.

**Review by hypevhs — APPROVED** [review]

**Pull request merged**